### DISCUSSION TITLE
Flytekit - Conversion of protobuf to python type files

### DISCUSSION THREAD

**Discussion by SandraGH5** [flyteorg/flyte#1377] [Q&A] [answered]
Quick Flytekit question. 
So I have an `input.pb` file and wanted to convert that into python type that can be given to task as input from a script.
So I am following the same method as in the source code: https://github.com/flyteorg/flytekit/blob/master/flytekit/common/tasks/executions.py#L40

but when i try to run ` _type_helpers.unpack_literal_map_to_sdk_python_std(input_map)` I got `TODO: Implement Map`
My `input.pb` does contain a dict with Dict[str, Flytefile] type.

How was this converted during actual execution? Is there another way to parse this to native python type?
I want to test an individual task with a given `input.pb` from a script, so is there an easy way to do this without me constructing the input itself?

- Jay Ganbat

**Comment by SandraGH5** [comment, marked as answer]
An `input.pb` is a `LiteralMap` so just follow this pattern since this is basically what the entrypoint does when a task runs for real.
https://github.com/flyteorg/flytekit/blob/3106c775fcc7425d38290062b5ab47a368fd448a/flytekit/bin/entrypoint.py#L102

So you can run `pyflyte-execute` locally - just pass the inputs as path on your local filesystem and outputs as another directory location, and also pass a raw output prefix on the local filesystem. Then see how `dispatch_execute` handles it.

Basically `pyflyte-execute` gets `inputs.pb` and parses it to give to the task.

@wild-endeavor 
@kumare3
